### DISCUSSION TITLE
Issues using cookiecutter template

### DISCUSSION THREAD

**Discussion by benwandrew** [AutoResearch/autora#402] [Core Contributions] [answered]
@younesStrittmatter, after running `cookiecutter https://github.com/AutoResearch/autora-firebase-prolific-cookiecutter` and answering the first two prompts (project name and python version), i selected the following example packages to include

<img width="662" alt="Screen Shot 2023-05-14 at 16 38 28" src="https://github.com/AutoResearch/autora/assets/28738595/f292bc13-f6ec-4ab9-b6ca-89f829c89356">

i keep receiving the following error when it proceeds to build my project

![Screen Shot 2023-05-14 at 16 50 00](https://github.com/AutoResearch/autora/assets/28738595/1cc2fbb5-5a21-4217-b7fd-092ec77f1bcc)

any ideas?

**Comment by younesStrittmatter** [comment]
I have (maybe) fixed the issue. Thanks a lot for testing, btw! It runs on my system, but there might be differences with other systems and I testing on other computers is invaluable.

**Comment by benwandrew** [comment]
awesome, the setup completed successfully!

but just to be sure, the initial project structure should just have a `researcher_environment` directory?

<img width="356" alt="Screen Shot 2023-05-14 at 17 14 12" src="https://github.com/AutoResearch/autora/assets/28738595/d5ff9c44-7498-40f5-b98b-c2dec9b65645">

also, sorry i'm still confused, but did we settle on underscores for names (eg, here it's `bwa_test_theorist`, although i specified `bwa-test-theorist` in response to the corresponding prompt)

**Reply by younesStrittmatter** [reply]
yes the cookiecutter template should set up a researcher_environment folder with a virtual environment. All the packages should be installed in that virtual environment. Folder always are set up with underscores, since the import would look like this: from bwa_test_theorist.sub_package import test_function. In the import statements dashes are not allowed therefore all folders that contain python code should be named with _.

**Reply by benwandrew** [reply]
oh right, of course 🤦

so far, so good then!

**Reply by younesStrittmatter** [reply]
I will add an example python script to the researcher_environment. The idea behind the "researcher_environment" is the following: Autora (at least for psychological research) has two "main" parts: The autora workflow, where the researchers live and work (researcher_environment with theorists/experimentalists/experiment_runner...) and the test_subject_environment where the participants get tested on (which will be the folder with the firebase stuff: node/javascript stuff for programming the experiment website). 

But maybe we can flatten the folder structure if the user doesn't want to use the Firebase stuff (so then no researcher_environment). Thanks, for pointing that out!

**Reply by younesStrittmatter** [reply]
Dashes and underscores are super confusing, for me, too. For example: If the folders are in the docs, you should use dashes instead, since they are showing up on a website and in an url the convention is to use dashes instead of underscores (so you can not just name all folders with underscores. Just the ones where python scripts are located)

**Reply by benwandrew** [reply]
> I will add an example python script to the researcher_environment. The idea behind the "researcher_environment" is the following: Autora (at least for psychological research) has two "main" parts: The autora workflow, where the researchers live and work (researcher_environment with theorists/experimentalists/experiment_runner...) and the test_subject_environment where the participants get tested on (which will be the folder with the firebase stuff: node/javascript stuff for programming the experiment website).
> 
> But maybe we can flatten the folder structure if the user doesn't want to use the Firebase stuff (so then no researcher_environment). Thanks, for pointing that out!

make sense! the only potential concern i have is using "environment" in the name of a directory which holds things other than development environments (eg, in this case, it will hold theorists/experimentalists/experiment_runner..., correct?) maybe that's just me, but is there something else we can call the directory where researchers live and work? or is "environment" the right call here?

**Reply by younesStrittmatter** [reply]
Yes. I would have called it autora_workflow, but that is overloaded as there is now a package called autora_workflow. But I am super open to suggestions. I also had the idea to call it something that resembles the "place" where reasearch is done (where researchers work), but to distinguish it from the place where the experiments are taken place, my only idea was "office" and I don't like that at all. Here is the appropriate place for naming suggestions: https://github.com/orgs/AutoResearch/discussions/365

**Comment by chadcwilliams** [comment]
I am getting a similar, but different error on Windows 10.

![Screenshot_2023-05-15 10_52_32](https://github.com/AutoResearch/autora/assets/26373111/c9f7ad55-8d18-4015-a8bf-65afa81b0e1c)

**Comment by younesStrittmatter** [comment, marked as answer]
I (hopefully) fixed the issue with the windows version and it should run smothely. Note that I also changed the name of the repository to autora-user-cookiecutter, so the command changed to cookiecutter https://github.com/AutoResearch/autora-user-cookiecutter
